rust/libnewsboat-ffi/src/utils.rs

```diff
@@ -170,6 +170,15 @@ pub unsafe extern "C" fn rs_censor_url(url: *const c_char) -> *mut c_char {
     })
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn rs_quote_for_stfl(string: *const c_char) -> *mut c_char {
+    abort_on_panic(|| {
+        let rs_string = CStr::from_ptr(string);
+        let rs_string = rs_string.to_string_lossy();
+        CString::new(utils::quote_for_stfl(&rs_string)).unwrap().into_raw()
```

Comment:
`unwrap()` aborts the whole application, so please write a comment explaining why this one can never fail.

Also, please run `cargo fmt` re-format this line: https://travis-ci.org/newsboat/newsboat/jobs/591447821

Reply:
Thanks for the review! I hope I've addressed your points with the new commits.

Did not know we had an rust autoformatter, but it seems nice and sensible :)

Hmm, I can see your point about adressing  ignored "Checked Exceptions" by using comments. I've added an comment on why I believe the NulError to not be thrown in that instance. Unfortunately the code will still blow up in an out-of-memory situation, but I do not think that Rust makes it really easy to catch this situation :/

Reply:
Hmm, I just realized however, that `cargo fmt` recommended a couple of changes to `rust/libnewsboat/src/fmtstrformatter/mod.rs` for my local runs... Does it have equal version-stability issues like clang-format has for C++?

Reply:
> Unfortunately the code will still blow up in an out-of-memory situation, but I do not think that Rust makes it really easy to catch this situation :/

Can be done on Nightly with String::try_reserve, but we're on stable, so it's impossible for us to handle that. If Newsboat runs out of memory, it will abort. Oh well.

Rustfmt just never gave any guarantees about non-ASCII strings. Their README says "Code containing non-ascii unicode characters (we believe Rustfmt mostly works here, but do not have the test coverage or experience to be 100% sure)". To be fair, clang-format isn't breaking their guarantees either—they just release major versions so often that it's indistinguishable from breaking the world all the time :)

Anyway, rustfmt breakage discussion can be continued in #654, and this PR is *done*.